torch/_inductor/codegen/wrapper.py

```diff
@@ -1754,13 +1754,22 @@ def generate_extern_kernel_out(self, output_view, codegen_reference, args, kerne
             self.writeline(self.wrap_kernel_call(kernel, args))
 
     def generate_user_defined_triton_kernel(self, kernel_name, grid, configs, args):
-        if len(grid) != 1:
-            raise NotImplementedError("triton.autotune not yet supported")
-        grid = grid[0]
+        assert len(grid) != 0
+        if len(grid) == 1:
+            grid_decision = grid[0]
+        else:
+            meta = CudaKernelParamCache.get(kernel_name)
+            assert meta is not None
+            grid_decision = None
+            for i, c in enumerate(configs):
+                if all(arg == meta[key] for key, arg in c.kwargs.items()):
+                    grid_decision = grid[i]
```

Comment:
Do we want to break from the loop if we get one valid choice?

Reply:
Or, being defensive, `assert grid_decision is None` before the assignment.